crates/core/src/client/mod.rs

```diff
@@ -611,7 +625,11 @@ impl<P: Provider + Send + Sync + 'static> KakarotStarknetApi<P> for KakarotClien
                 let block = self.starknet_provider.get_block_with_tx_hashes(block_id).await?;
                 match block {
                     MaybePendingBlockWithTxHashes::Block(block_with_tx_hashes) => Ok(block_with_tx_hashes.block_number),
-                    _ => Err(ProviderError::StarknetError(StarknetError::BlockNotFound).into()),
+                    _ => Err(ProviderError::StarknetError(StarknetErrorWithMessage {
+                        code: MaybeUnknownErrorCode::Known(StarknetError::BlockNotFound),
+                        message: "".to_string(),
```

Comment:
why are we leaving an empty string here?

Reply:
Main reason is that it isn't really used, since in the `errors.rs`, we would just convert `BlockNotFound` to string. The message is only used in the case of a `MaybeUnknownErrorCode::Unknown`. I can add a message if you want but if fells wasteful